Sources/NIOKit/ConnectionPool.swift

```diff
@@ -83,8 +83,9 @@ public final class ConnectionPool<Source> where Source: ConnectionPoolSource  {
         self.config = config
         self.source = source
         self.available = []
+        self.available.reserveCapacity(config.maxConnections)
```

Comment:
Out of curiosity, why did we add this? :)

Reply:
The array should never need to hold more than that capacity, so reserving should prevent any need for array re-alloc during usage.

Reply:
Thought that was it, thanks!